src/saveload/ser.rs

```diff
@@ -0,0 +1,169 @@
+                    None => {
+                        let marker = allocator.mark(entity, markers);
+                        add.push((entity, marker));
+                        Some(marker.id())
+                    }
+                }
+            };
+            for (entity, marker) in to_serialize.into_iter() {
+                serseq.serialize_element(&EntityData::<M, E, T> {
+                    marker: marker,
+                    components: T::save(entity, storages, &mut ids).map_err(ser::Error::custom)?,
+                })?;
+            }
+        }
+        to_serialize = add;
+    }
+    serseq.end()
+}
+
+
+/// Serialize components from specified storages via `SerializableComponent::save`
+/// of all marked entities with provided serializer.
+/// When the component gets serialized with `SerializableComponent::save` method
+/// closure passed in `ids` arguemnt returns `None` for unmarked `Entity`.
+/// In this case `SerializableComponent::save` may perform workaround (forget about `Entity`)
+/// or fail.
+/// So function doesn't recursively mark referenced entities.
+/// For recursive marking see `serialize_recursive`
+pub fn serialize<'a, M, E, T, S>(
+    entities: &Entities<'a>,
+    storages: &<T as Storages<'a>>::ReadStorages,
+    markers: &ReadStorage<'a, M>,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    M: Marker,
+    E: Display,
+    T: Components<M::Identifier, E>,
+    S: Serializer,
+{
+    let mut serseq = serializer.serialize_seq(None)?;
+    let ids = |entity| -> Option<M::Identifier> { markers.get(entity).map(Marker::id) };
+    for (entity, marker) in (&**entities, &*markers).join() {
+        serseq.serialize_element(&EntityData::<M, E, T> {
+            marker: *marker,
+            components: T::save(entity, storages, &ids).map_err(ser::Error::custom)?,
+        })?;
+    }
+    serseq.end()
+}
+
+/// This type implements `Serialize` so that it may be used in generic environment
+/// where `Serialize` implementer is expected.
+/// It may be constructed manually with `WorldSerialize::new`.
+/// Or fetched by `System` as `SystemData`.
+/// Serializes components in tuple `T` with marker `M`.
+#[derive(SystemData)]
+pub struct WorldSerialize<'a, M: Marker, E, T: Components<M::Identifier, E>> {
+    entities: Entities<'a>,
+    storages: <T as Storages<'a>>::ReadStorages,
+    markers: ReadStorage<'a, M>,
+    pd: PhantomData<E>,
+}
+
+macro_rules! world_serialize_new_functions {
+    ($($a:ident),*) => {
+        impl<'a, X, Z $(,$a)*> WorldSerialize<'a, X, Z, ($($a,)*)>
+            where X: Marker,
+                $(
+                    $a: super::SaveLoadComponent<X::Identifier>,
+                    Z: From<$a::Error>,
+                )*
+        {
+            /// Create serializable structure from storages
+            #[allow(non_snake_case)]
+            pub fn new(entities: Entities<'a>,
+                       markers: ReadStorage<'a, X>
```

Comment:
nit: please use the new formatting

Reply:
This is not resolved, could you use the latest version of rustfmt-nightly and format this please?